feed/views.py

```diff
@@ -1,60 +1,184 @@
-                    'product_image'  : feed.product.productimageurl_set.get(is_main=1).image_url
-                    } if feed.product_id else {
-                    'product_id'     : '',
-                    'product_name'   : '',
-                    'discount_rate'  : '',
-                    'price'          : '',
-                    'product_image'  : ''
-                    },
-
-                'feed_comment_data'  : {
-                    'feed_comment_count' : feed.comment_set.count(),
-                    'comment_list'       : [{
-                            'user'      : User.objects.get(id = comment.user_id).nickname,
-                            'content'   : comment.content
-                        } for comment in feed.comment_set.all().order_by('-created_at')[:MAXIMUM_COMMENT]
-                        ] if feed.comment_set.exists() else
-                        [{
-                            'user':'',
-                            'content':''
-                        }]
+                    'product_data'   : {
+                        'product_id'     : feed.product_id,
+                        'product_name'   : feed.product.name,
+                        'discount_rate'  : feed.product.discount_rate,
+                        'price'          : feed.product.price,
+                        'product_image'  : feed.product.productimageurl_set.get(is_main=1).image_url
+                        } if feed.product_id else {
+                        'product_id'     : '',
+                        'product_name'   : '',
+                        'discount_rate'  : '',
+                        'price'          : '',
+                        'product_image'  : ''
+                        },
+    
+                    'feed_comment_data'  : {
+                        'feed_comment_count' : feed.comment_set.count(),
+                        'comment_list'       : [{
+                                'user'      : User.objects.get(id = comment.user_id).nickname,
+                                'content'   : comment.content
+                            } for comment in feed.comment_set.all().order_by('-created_at')[:MAXIMUM_COMMENT]
+                            ] if feed.comment_set.exists() else
+                            [{
+                                'user':'',
+                                'content':''
+                            }]
+                        }
+                    }
+    
+                feed_list.append(feed_data)
+            feed_list = feed_list[offset:offset+limit]
+    
+            return JsonResponse({'feed_list': feed_list}, status=200)
+
+        except User.DoesNotExist:
+            return JsonResponse({'MESSAGE' : 'INVALID_USER'}, status=404)
+
+class FeedDetailView(View):
+    def get(self, request, feed_id):
+        try:
+            feed_data   = Feed.objects.get(id=feed_id)
+            feed_writer = User.objects.get(id=feed_data.user_id)
```

Comment:
```suggestion
            feed_writer = feed_data.user
```
바로 위에서 `feed_data` 객체가 있기 때문에 feed_data의 속성에서 꺼내오시는 것을 추천드립니다

Reply:
위의 리뷰와 지금 이 리뷰를 통해 .user_id로 fk 값을 받아오는 것 말고 .user로 객체도 받을 수 있다는 걸 알았어요. 감사합니다.